samples/sample-web-wallet-app/src/screens/MainScreen.tsx

```diff
@@ -107,16 +108,47 @@ const onSubmitPresentation = (
   presentationRequest: Dictionary<any>,
   authToken?: Dictionary<any>
 ) => {
-  submitPresentation({
-    verifiableCredentials: Constants.getIdentificationList(environment),
-    presentationRequest,
-    authToken,
-  })
+  let authTokenRefreshAmount = 0;
```

Comment:
the logic should be
1. inspect the accessToken and check if its expired - request the new token (abort should bubble up).
2. SubmitPresentation using accessToken
3. loop if it errors and try again

Get token logic
1. Request token
2. If token requesting fails then abort.


Ideally this will allow you to use async/await